src/util/container-request.js

```diff
@@ -0,0 +1,51 @@
+import {detectContainerForBytes, getId3Offset} from '@videojs/vhs-utils/dist/containers';
+import {stringToBytes, toUint8} from '@videojs/vhs-utils/dist/byte-helpers';
+import {callbackWrapper} from '../xhr';
+
+const containerRequest = (uri, xhr, cb) => {
+  let bytes;
+  let id3Offset;
+
+  const progressListener = function(error, request) {
+    if (error) {
+      return cb(error, request, '', bytes);
+    }
+    const currentLength = bytes && bytes.length || 0;
+    const contentPart = request.responseText.substring(currentLength, request.responseText.length);
+    const newBytes = toUint8(stringToBytes(contentPart, true));
+    const temp = new Uint8Array(currentLength + newBytes.length);
+
+    if (bytes) {
+      temp.set(bytes);
+    }
+    temp.set(newBytes, currentLength);
+    bytes = temp;
+
+    if (!id3Offset) {
+      id3Offset = getId3Offset(bytes);
+    }
+
+    // not enough data to determine type yet
+    if (bytes.length < id3Offset + 2) {
+      return;
+    }
+    request.abort();
+    const type = detectContainerForBytes(bytes);
```

Comment:
If there's no id3 offset, and bytes.length is 3, would that be enough bytes to determine the container in all cases?

Reply:
Only mp3 and aac should have id3 offset and they only need two bytes after the id3 data to determine the type

Reply:
I'm probably missing something, but I guess what I'm asking is, if we're trying to detect the container of a video mp4 segment, and we need 4 bytes for that detection, but we got 3 bytes on progress, will this return no detected type and the playlist get blacklisted?

Reply:
I think you may have a point her, going to refactor this a bit.